__Question 1__: Given some sample data, write a program to answer the following: click here to access the required data set

On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis. 


In [1]:
import pandas as pd

# First we will read in the dataset and take a look at it

In [2]:
df = pd.read_csv('ShopifySneakers.csv')

In [9]:
df.head(20)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0          1       53      746           224            2           cash   
1          2       92      925            90            1           cash   
2          3       44      861           144            1           cash   
3          4       18      935           156            1    credit_card   
4          5       18      883           156            1    credit_card   
5          6       58      882           138            1    credit_card   
6          7       87      915           149            1           cash   
7          8       22      761           292            2           cash   
8          9       64      914           266            2          debit   
9         10       52      788           146            1    credit_card   
10        11       66      848           322            2    credit_card   
11        12       40      983           322            2          debit   
12        13       54      799           266            2    credit_card   
13        14      100      709           111            1           cash   
14        15       87      849           447            3    credit_card   
15        16       42      607        704000         2000    credit_card   
16        17       17      731           176            1           cash   
17        18       28      752           164            1    credit_card   
18        19       83      761           258            2           cash   
19        20       63      898           408            3    credit_card   

             created_at  
0   2017-03-13 12:36:56  
1   2017-03-03 17:38:52  
2    2017-03-14 4:23:56  
3   2017-03-26 12:43:37  
4    2017-03-01 4:35:11  
5   2017-03-14 15:25:01  
6   2017-03-01 21:37:57  
7    2017-03-08 2:05:38  
8   2017-03-17 20:56:50  
9   2017-03-30 21:08:26  
10  2017-03-26 23:36:40  
11  2017-03-12 17:58:30  
12  2017-03-16 14:15:34  
13   2017-03-22 2:39:49  
14  2017-03-10 11:23:18  
15   2017-03-07 4:00:00  
16   2017-03-21 4:23:38  
17  2017-03-21 12:09:07  
18  2017-03-17 13:18:47  
19  2017-03-29 15:11:52

In [12]:
# To confirm the average number, we will calculate the average of order_amount

average_order_value = df['order_amount'].mean()
print(round(average_order_value, 2))

3145.13

Analyzing this dataframe, we see that this average order value is exceedingly high. However, there exists a column call total_items, which is a count for the number of items sold per order. This shows us that customers are buying, at times, more than 1 item per order, giving us this inflated value. In fact, we can see that for order_id 15, we have an order amount of $704000! This value is very high but looking at total_items, we see that this customer actually bought 2000 pairs from this store. We can use the sum of the total_items divided by the total order_amount value to calculate average __item__ value instead of average __order__ value.

In [11]:
average_item_value = df['order_amount'].sum()/df['total_items'].sum()
print(round(average_item_value, 2))

357.92


We get an average item value of $357.92 which, although may be high for some people for a pair of sneakers, is much more reasonable as an answer. 

__Question 2__: For this question you’ll need to use SQL. Follow this link to access the data set required for the challenge. Please use queries to answer the following questions. Paste your queries along with your final numerical answers below.

a. How many orders were shipped by Speedy Express in total?

Answer: 54 orders were shipped by Speedy Express in total

    SELECT * FROM [Orders]
    JOIN Shippers
    ON Orders.ShipperID = Shippers.ShipperID
    WHERE ShipperName = 'Speedy Express'

b. What is the last name of the employee with the most orders?

Answer: Handel (Customer ID = 20)

    SELECT *, COUNT(Orders.CustomerID) FROM Orders
    JOIN Customers
    ON Customers.CustomerID = Orders.CustomerID
    GROUP BY Orders.CustomerID
    ORDER BY COUNT(Orders.CustomerID) DESC;

c. What product was ordered the most by customers in Germany?

Answer: Boston Crab Meat (160 units)

    SELECT OrderDetails.ProductID, OrderDetails.OrderID, Orders.OrderID, Customers.CustomerID, Orders.CustomerID, Country, ProductName, SUM(Quantity) FROM OrderDetails, Orders, Customers, Products
    WHERE OrderDetails.OrderID = Orders.OrderID
    AND Customers.CustomerID = Orders.CustomerID
    AND OrderDetails.ProductID = Products.ProductID
    AND Country = 'Germany'
    GROUP BY OrderDetails.ProductID
    ORDER BY SUM(Quantity) DESC;
    
(SQL Knowledge is still a WIP.)